UnicornRender/video/source/Mesh.cpp

```diff
@@ -10,11 +10,16 @@ namespace unicorn
 {
 namespace video
 {
-Mesh::Mesh(Material material)
-    : m_material(material)
+Mesh::Mesh() :
+    m_material(new Material)
```

Comment:
this behaves differently from `SetMaterial()` I think you should instead initialize with `nullptr` and call `SetMaterial()` since you have a complex logic of binding to material